cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -948,11 +961,16 @@ func (s *Supervisor) composeMergedConfig(config *protobufs.AgentRemoteConfig) (c
 	}
 
 	// Check if supervisor's merged config is changed.
-	newMergedConfig := string(newMergedConfigBytes)
+
+	newConfigState := &configState{
+		mergedConfig:   string(newMergedConfigBytes),
+		emptyConfigMap: len(config.GetConfig().GetConfigMap()) == 0,
```

Comment:
I'm curious about the difference between this condition and the `if config.GetConfig() != nil` above. Are we essentially evaluating the same thing twice, or is there a meaningful difference. If they are not the same, are we sure at this point that `config.GetConfig()` is not nil? (I don't see where it would have changed, but maybe I missed it.) If they are the same, should we evaluate it once, then use it to control the if/else and also to set this field?

Reply:
I think you're right that the nil check and this check should be unified in some way. I do think there is currently a subtle difference between the two, but this bool (emptyConfigMap) is a more general and basically overrides whatever is being done above anyways. I'll take a look at that.

Reply:
I tried to unify them here. A note about chaining these getters from the generated protobufs - all these getters have nil checks in them that avoid panic here. So if the `config.GetConfig()` returns  a nil *AgentConfigMap, calling GetConfigMap on the nil value doesn't panic, it just returns a nil map (which has size 0).